src/main/java/org/opensrp/web/rest/EventResource.java

```diff
@@ -545,5 +677,24 @@ public void setEventService(EventService eventService) {
 	public void setClientService(ClientService clientService) {
 		this.clientService = clientService;
 	}
-	
+
+	public void setMultimediaService(MultimediaService multimediaService) {
+		this.multimediaService = multimediaService;
+	}
+
+	private void generateCSV(ExportEventDataSummary exportEventDataSummary, URI uri) throws IOException {
+
+		BufferedWriter writer = Files.newBufferedWriter(Paths.get(uri));
+		CSVPrinter csvPrinter = new CSVPrinter(writer, CSVFormat.DEFAULT);
+
+		for (List<Object> rows : exportEventDataSummary.getRowsData()) {
+			csvPrinter.printRecord(rows);
+			csvPrinter.printRecord("\n");
+		}
+		csvPrinter.flush();
+		writer.close();
+		csvPrinter.close();

```

Comment:
close in finally block or use try with resources

Reply:
Done